src/components/Menu/MenuItem/hooks/useMenuItemMouseEvents.ts

```diff
@@ -18,10 +20,37 @@ export default function useMenuItemMouseEvents({
   setActiveItemIndex: (index: number) => void;
   index: number;
   hasChildren: boolean;
+  splitMenuItemIconButtonRef?: React.RefObject<HTMLElement>;
+  splitMenuItem?: boolean;
 }) {
   const isMouseEnter = useIsMouseEnter({ ref });
-
+  const isMouseEnterOnIconButton = useIsMouseEnter({ ref: splitMenuItemIconButtonRef });
   const prevIsMouseEnter = usePrevious(isMouseEnter);
+  const prevIsMouseEnterOnIconButton = usePrevious(isMouseEnterOnIconButton);
+
+  useLayoutEffect(() => {
```

Comment:
Sub menu doesn't close on remove hover from the IconButton

Reply:
yes i know, we tried to make it possible and I sat with Yossi and Tal for hours but we couldn't do it. the render cycle of menu item happens 5 times for each action, and there are a lot of elements in menu item that opens sub menu - menu item, icon button, submenu dialog, submenu menu item. because of these two reasons, we couldn't control the mouse behavior. I approved this behavior with Meirav.

Reply:
Was it rerendering like this before also or is it new behaviour?

Reply:
we never could make it work...i'm not sure what you mean by new behavior? this is a new type of menu item